cli/deepsea.py

```diff
@@ -119,80 +122,144 @@ def _validate_stage_file_exists(stage_name):
         sys.exit(1)
 
 
-def _run_show_stage_steps(stage_name, hide_state_steps, only_visible_steps, use_cache):
+def _print_stage_step(step, indent, step_order_map):
+    if isinstance(step, SaltState):
+        PP.print(PP.orange(step.sls))
+        PP.print(PP.dark_green(" ({})".format(step.desc)))
+        PP.println(PP.orange(" on"))
+        if step.on_success_deps or step.on_fail_deps:
+            if step.on_success_deps:
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_success_deps]
+                deps.sort()
+                PP.print(PP.dark_yellow("if_success=[{}] ".format(", ".join(deps))))
+            if step.on_fail_deps:
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_fail_deps]
+                deps.sort()
+                PP.println(PP.dark_yellow("if_fail=[{}]".format(", ".join(deps))))
+            else:
+                PP.println()
+
+        for minion in step.target:
+            PP.print("{}".format(" " * indent))
+            PP.println(PP.cyan(minion))
+            step_count = 1
+            for s_step in step.steps[minion]:
+                PP.print("{}|_ ".format(" " * (indent + 2)))
+                if s_step.visible:
+                    PP.print(PP.green("+ "))
+                else:
+                    PP.print(PP.red("- "))
+                PP.print(PP.bold("{:3} ".format("{}.".format(step_count))))
+                step_order_map[id(s_step)] = str(step_count)
+                _print_stage_step(s_step, indent + 11, step_order_map)
+                step_count += 1
+
+    elif isinstance(step, SaltRunner):
+        PP.print(PP.blue(step.function))
+        PP.println(PP.dark_green(" ({})".format(step.desc)))
+        if step.on_success_deps or step.on_fail_deps:
+            if step.on_success_deps:
+                PP.print("{}".format(" " * indent))
```

Comment:
Since we use the same code here as above, does it make sense to pull it out into a method?
I am thinking one method for `step.on_success_deps` and one for `step.on_fail_deps` that takes care of the printing.